FluidNC/src/Configuration/RuntimeSetting.cpp

```diff
@@ -72,6 +72,22 @@ namespace Configuration {
         }
     }
 
+    void RuntimeSetting::item(const char* name, uint32_t& value, uint32_t minValue, uint32_t maxValue) {
+        if (is(name)) {
+            isHandled_ = true;
+            if (newValue_ == nullptr) {
+                out_ << "$/" << setting_ << "=" << value << '\n';
+            } else {
+                if (newValue_[0] == '-') {  // constrian negative values to 0
+                    value = 0;
+                    log_warn("Negative value not allowed");
+                } else
+                    value = atoll(newValue_);
+            }
+            constrain_with_message(value, minValue, maxValue);
```

Comment:
The canonical way to do error checking is to call strtoull with a non-null "endptr" argument.  Then check to see if the value returned in endptr is the same as startptr + strlen(startptr).  That will handle all manner of invalid characters.